### PR TITLE
feat: add search api to leads

### PR DESCRIPTION
**Pull request by djanov**
References [CAR-5982](https://autoricardo.atlassian.net/browse/CAR-5982)

## Motivation and context
add for search api for the leads

### CODE REVIEW

File: src/types/params/types.ts [hunk 1]
```diff
@@ -21,6 +21,9 @@ export interface TypeFiltersParams {
   productionYear?: number
   imagesCountGroup?: string[]
 }
+export interface LeadTypeFiltersParams {
+  searchQuery?: string
```
Comment (masone) [inline src/types/params/types.ts:25]: what is this required for? wouldn't one use `LeadQueryParams` instead? this type is not used anywhere, but only exported

Comment (djanov) [reply src/types/params/types.ts:25]: I added that like the `TypeFiltersParams` and use it in DH, but I just checked I don't really use it for the search but for the `filters `and `derivedFilters` in the params.ts, but we will need this later, but for now I will delete here and on DH too.

File: src/types/params/leads.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { LeadSortTypeParams, SortParams } from "../../types/sort"
 
 import { PaginationParams } from "./index"
-
 export interface LeadQueryParams extends PaginationParams {
   sort: SortParams<LeadSortTypeParams>
+  searchQuery: string
```
Comment (lkappeler) [inline src/types/params/leads.ts:6]: Should the param be mandatory? Look like your are defaulting it in the search function...

Comment (djanov) [reply src/types/params/leads.ts]: @lkappeler you are right, it should be optional, my bad 🤦‍♂️

**Review by lkappeler — APPROVED** [review]

**Pull request merged**